src/traceradvectiondiffusion.jl

```diff
@@ -230,6 +319,30 @@ function calcN_steadyflow!(N, sol, t, clock, vars, params::AbstractSteadyFlowPar
   return nothing
 end
 
+"""
+    calcN_turbulentflow!(N, solt, t, clock, vars, params::AbstractTurbulentFlowParams, grid)
+
+Calculate the advective terms for a tracer equation with constant diffusivity and turbulent `MultiLayerQG` flow.
+"""
+function calcN_turbulentflow!(N, sol, t, clock, vars, params::AbstractTurbulentFlowParams, grid)
+    @. vars.cxh = im * grid.kr * sol
+    @. vars.cyh = im * grid.l  * sol
+  
+    MultiLayerQG.invtransform!(vars.cx, vars.cxh, params.MQGprob.params)
+    MultiLayerQG.invtransform!(vars.cy, vars.cyh, params.MQGprob.params)
+
+    u = @. params.MQGprob.vars.u + params.MQGprob.params.U
+    v = params.MQGprob.vars.v
+
+    # Step the flow forward for next iteration
+    MultiLayerQG.stepforward!(params.MQGprob)
```

Comment:
this implies that MultiLayerQG problem is stepped forward every time `calcN_turbulentflow!` is called. But in, e.g., an RK4 time stepping scheme `calcN_turbulentflow!` will be called 4 times for each tilmestep.

Reply:
I suggest that `MultiLayerQG.stepforward!(params.MQGprob)` is called outside the `calcN_turbulentflow!` function. Inside `calcN_turbulentflow!` the u and v velocities from MQGprob should be obtained and then used to advect the tracer.

Reply:
```Julia
julia> ADprob.params.
MQGprob             nlayers              nκh                  tracer_release_time  η                    κ                    κh
julia> ADprob.params.MQGprob.clock.step
24005
```

Reply:
With `tracer_time_release = 8000 * dt` one would expect the latter to be just 8000 more. But apparently
```
ADprob.params.MQGprob.clock.step = 8000 + 4 * ADprob.clock.step
```
which happens because of the 4 times per tilmestep that `MultiLayerQG.stepforward!` is called.

Reply:
Ah I see. My fault! Will fix this up now.

Reply:
For now I have put `MultiLayerQG.stepforward!(params.MQGprob)` and `MultiLayerQG.updatevars!(params.MQGprob)` in the example. Ideally these will be called when `stepforward!(ADprob)` but not quite sure of the  best way to do this right now.